content/matrices/components/gauss-solver.ts

```diff
@@ -0,0 +1,196 @@
+
+
+import {$N, CustomElementView, ElementView, Observable, observe, register, slide} from '@mathigon/boost';
+import {tabulate2D} from '@mathigon/core';
+import {Point} from '@mathigon/euclid';
+import {clamp} from '@mathigon/fermat';
+import {Expression} from '@mathigon/hilbert';
+import {Step} from '../../shared/types';
+
+
+import template from './gauss-solver.pug';
+
+type Model = {
+  input: number[][], output: number[][], op: string, size: number,
+  inRow1: number, inRow2: number, outRow1: number, outRow2: number,
+  factorString: string, factor: number};
+
+@register('x-gauss-solver', {template})
+export class GaussSolver extends CustomElementView {
+  model!: Observable<Model>;
+  size!: number; // Number of Matrix rows
+  $inputCells!: ElementView[][]; // Array of all cells in the left matrix.
+  $outputCells!: ElementView[][]; // Array of all cells in the right matrix.
+  $step!: Step;
+  inputStack!: string[][][];
+
+  ready() {
+    const input = Expression.parse(this.attr('matrix'))
+        .evaluate({'[': (...args: number[]) => [...args] as any}) as unknown as number[][];
+
+    this.size = input.length;
+    this.bindModel(observe({input, output: input, size: this.size, inRow1: 0, inRow2: 1, outRow1: 0, outRow2: 1, factor: 1, factorString: '1'}));
+
+    // Set up the input and output matrices
+    const $matrices = this.$$('.matrix');
+    for (const $m of $matrices) $m.css('grid-template-columns', `repeat(${this.size + 1}, 36px)`);
+
+    this.$inputCells = tabulate2D((i, j) =>
+      $N('div', {text: input[i][j]}, $matrices[0]), this.size, this.size + 1
+    );
+    this.$outputCells = tabulate2D(() => $N('div', {text: 0}, $matrices[1]), this.size, this.size + 1);
+
+    this.inputStack = [this.$inputCells.map(row => row.map(val => val.text))];
+
+    const $circles = this.$$('.connections-left circle');
+    const inRows = ['inRow1', 'inRow2'] as ('inRow1'|'inRow2')[];
+
+    for (const [i, key] of inRows.entries()) {
+      slide($circles[i], {
+        move: (p: Point) => {
+          // Check that this row is not equal to the other input
+          // ^ I removed this feature because it makes it difficult w/ 2x2 matrix to choose which one you want to multiply
+          const row = clamp(Math.round((p.y - 20) / 40), 0, this.size - 1);
+          /* if (this.model.op === 'multiply' || this.model[inRows[i === 0 ? 1 : 0]] !== row)*/ this.model[key] = row;
+        }
+      });
+    }
+
+    const $arrows = this.$$('.connections-right polygon');
+    const outRows = ['outRow1', 'outRow2'] as ('outRow1'|'outRow2')[];
+
+    for (const [i, key] of outRows.entries()) {
+      slide($arrows[i], {
+        move: (p: Point) => {
+          // Check that this row not equal to the other output
+          const row = clamp(Math.round((p.y - 20) / 40), 0, this.size - 1);
+          this.model[key] = row;
```

Comment:
I don't think the output rows need to be interactive right? They are the same (or swapped) as the input rows